bundles/action/org.openhab.action.pushbullet/src/main/java/org/openhab/action/pushbullet/internal/PushbulletAPIConnector.java

```diff
@@ -126,7 +133,16 @@ private static boolean sendPush(String botName, String recipient, String title,
         push.setTitle(title);
         push.setBody(body);
         push.setType(type);
-        push.setEmail(recipient);
+
+        if (recipient != null) {
+            if (isValidEmail(recipient)) {
+                logger.trace("    Recipient is an email address");
+                push.setEmail(recipient);
+            } else if (isValidChannel(recipient)) {
+                logger.trace("    Recipient is a channel tag");
+                push.setChannel(recipient);
+            }
```

Comment:
else  ?

What happens if neither of these is satisfied? Shouldn't this then throw?

Reply:
If neither of these is satisfied, no specific target will be included in the Pushbullet API query.

Per the [API documentation](https://docs.pushbullet.com/#target-parameters), if no target is specified, the notification will be broadcast to all of the user's devices, which is considered as the default behavior.

Reply:
I added an log warn to let the user know that the provided recipient was invalid.